etna/articles/migrations/0056_set_teaser_text_from_intro.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 4.0.8 on 2023-02-16 16:40
+
+from django.db import migrations
+from django.db.models import F
+
+
+PAGE_MODEL_NAMES = (
+    "articles.ArticleIndexPage",
+    "articles.ArticlePage",
+)
+
+
+def migrate_forwards(apps, schema_editor):
+    for model_name in PAGE_MODEL_NAMES:
+        model_class = apps.get_model(*model_name.split("."))
+
+        # update the page objects themselves
+        model_class.objects.update(teaser_text=F("intro"))
+
+        # update latest revision content
+        for page in model_class.objects.all().select_related("latest_revision"):
+            rev = page.latest_revision
+            content = rev.content
+            if "intro" not in content:
+                content["intro"] = content.get("sub_heading", "")
+            content["teaser_text"] = content["intro"]
```

Comment:
Thought: This might be my understanding but the latest revision doesn't include the live data if a user never saves it as a draft? This may miss content that is published if that is the case.